### ISSUE TITLE
Option to calculate the Centre Line as an Average or Median

### ISSUE THREAD

**Issue by JagDhami1** [AUS-DOH-Safety-and-Quality/PowerBI-SPC#273]
Hi, at the NHS we tend to use the median as the centre line in our SPC charts with upper and lower control limits. The Run Chart in the visual seems to calculate the centre line as the Mean, but the i chart (Individual Measurements) seems to calculate the centre line as the Average.

Could you add an option to the 'Data Settings' where the centre can be calculated as the Mean/Average. Even better if there was an option to enter a function (fx) to calculate the centre line.

Many thanks
Jag.

**Comment by JagDhami1** [comment]
Thanks for your reply and we would certainly appreciate additional chart types for these.

**Comment by andrjohns** [comment]
The development version of the visual now has an option for i chart with medians:

![image](https://github.com/AUS-DOH-Safety-and-Quality/PowerBI-SPC/assets/27717896/d5958a12-12cf-46ae-958b-28219b04b963)

**Comment by JagDhami1** [comment]
Many thanks Andrew, I will download and test.

**Comment by JagDhami1** [comment]
Hi Andrew, thank you for adding the new chart type which looks perfect. However, when I add the 99% and 95% control limits the plotted lines don't seem representative of the data points. I would expect the lines to be a lot broader.

![image](https://github.com/AUS-DOH-Safety-and-Quality/PowerBI-SPC/assets/166033455/e60a6b90-9407-4082-a181-db5cbfa15835)

and more like this.

![image](https://github.com/AUS-DOH-Safety-and-Quality/PowerBI-SPC/assets/166033455/26361694-72f3-4a98-bbcc-f6e3cd0a3527)

**Comment by andrjohns** [comment]
Hmm there's definitely something wrong there. Are you able to share a dataset that reproduces this? I don't get the same discrepancy with the datasets I've tested, so it's hard to say where it might be coming from

**Comment by JagDhami1** [comment]
Yes off course, this is the dataset I'm using
<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/jdha00/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/jdha00/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">


Value | Date
-- | --
16 | 06/03/2024 00:00
22 | 07/03/2024 00:00
22 | 08/03/2024 00:00
26 | 09/03/2024 00:00
28 | 10/03/2024 00:00
0 | 11/03/2024 00:00
30 | 12/03/2024 00:00
28 | 13/03/2024 00:00
28 | 14/03/2024 00:00
22 | 15/03/2024 00:00
24 | 16/03/2024 00:00
26 | 17/03/2024 00:00
22 | 18/03/2024 00:00
23 | 19/03/2024 00:00
22 | 20/03/2024 00:00
16 | 21/03/2024 00:00
16 | 22/03/2024 00:00
16 | 23/03/2024 00:00
24 | 24/03/2024 00:00
16 | 25/03/2024 00:00
16 | 26/03/2024 00:00
16 | 27/03/2024 00:00
16 | 28/03/2024 00:00
16 | 29/03/2024 00:00
16 | 30/03/2024 00:00
16 | 31/03/2024 00:00
24 | 01/04/2024 00:00



</body>

</html>

**Comment by andrjohns** [comment]
Interestingly, the chart is actually doing the correct thing here. Your process mostly has little to no change between subsequent months, with a small amount of large changes. This leads to a large discrepancy between the average moving range (used to calculate the normal i-chart limits) and the median moving range:

- Average MR = 4.54
- Median MR = 2

Additionally, the limit calculations for i-charts include a step which filters outlying moving-ranges and then re-calculates the mean/median moving range (you can disable this using the `Keep Outliers in Limit Calcs.` option underneath the chart type dropdown), this then leaves a median MR of 1.

In other words, this indicates that your process usually shows very small changes, with a small amount of large changes.

But I can see how this level of sensitivity might not be useful in this case, so I've also changed the input options so that you can select median centerline with or without median moving-range limits:

![image](https://github.com/AUS-DOH-Safety-and-Quality/PowerBI-SPC/assets/27717896/e9c25435-9b53-4677-b14b-4ffd203b5a26)

**Comment by JagDhami1** [comment]
Thank you again Andrew for your detailed explanation and the new chart type. I feel like I'm only highlighting the negative parts of the vis, but rest assured this will be a very important visual tool for the NHS. Thanks!
